### PR TITLE
Regression: Permissions missing on new Room Edit and Contact Edit form

### PR DESCRIPTION
**Pull request by murtaza98**

### CODE REVIEW

**Review by renatobecker — CHANGES_REQUESTED** [review]
The `Edit Contact` action works like a charm, but the `Edit Room` action doesn't. This is what I get:

Screen Shot 2021-03-26 at 10 59 55 AM

**Comment by murtaza98** [comment]
> The `Edit Contact` action works like a charm, but the `Edit Room` action doesn't. This is what I get:
> 
> Screen Shot 2021-03-26 at 10 59 55 AM

@renatobecker Sorry I wasn't able to reproduce this error. Could you please check again ?

**Comment by renatobecker** [comment]
@murtaza98 the implementation fixes the issues, however, it's clear that both `Contact` and `Room` contextual bar are placed in the wrong folder ->   `client/omnichannel/directory/`.

Since those components are used in different areas of the product, they should be placed in a "global" folder, not inside the "directory" folder.

@rafaelblink please, help us reviewing this PR. Thanks.

**Comment by murtaza98** [comment]
> @murtaza98 the implementation fixes the issues, however, it's clear that both `Contact` and `Room` contextual bar are placed in the wrong folder -> `client/omnichannel/directory/`.
> 
> Since those components are used in different areas of the product, they should be placed in a "global" folder, not inside the "directory" folder.
> 
> @rafaelblink please, help us reviewing this PR. Thanks.

@renatobecker I've moved out the files. Could you please review it again? Thanks

**Review by renatobecker — COMMENTED** [review]
Where are we checking permissions for room editing and contact editing? I didn't find it.
This is how we used to check it on the old version:

https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/views/app/tabbar/visitorInfo.js#L184-L189

Also, we can't just add a permission check for rendering the button or not because the user can access the route through the URL, so the permission needs to allow/deny accessing the edit form(room or contact) anyways.

**Comment by murtaza98** [comment]
@renatobecker For Room i've implemented the same logic as [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/views/app/tabbar/visitorInfo.js#L184-L189).

However I'm not sure what permission to use for Contact edit? Should I create a new global permission `edit-contacts` and by default apply this permission to agents and managers?

**Comment by renatobecker** [comment]
> Should I create a new global permission `edit-contacts`

`edit-omnichannel-contact`

> and by default apply this permission to agents and managers?

Yes.

**Comment by murtaza98** [comment]
> > Should I create a new global permission `edit-contacts`
> 
> `edit-omnichannel-contact`
> 
> > and by default apply this permission to agents and managers?
> 
> Yes.

Hey @renatobecker, This part is complete

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: app/authorization/server/startup.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ Meteor.startup(function() {
 		{ _id: 'call-management',                    roles: ['admin', 'owner', 'moderator'] },
 		{ _id: 'create-invite-links',                roles: ['admin', 'owner', 'moderator'] },
 		{ _id: 'view-l-room',                        roles: ['livechat-agent', 'livechat-manager', 'admin'] },
-		{ _id: 'view-livechat-manager',              roles: ['livechat-manager', 'admin'] },
```
[the remark this answers is not in the document]
Comment (murtaza98) [reply app/authorization/server/startup.js:95]: Sorry that was by mistake. Glad you caught it :raised_hands:

File: app/ui/client/views/app/room.js [hunk 2]
```diff
@@ -890,7 +890,7 @@ Meteor.startup(() => {
 
 			let room = Rooms.findOne({ _id: rid }, { fields: { t: 1 } });
 
-			if (room?.t === 'l' && !FlowRouter.getParam('tab')) {
+			if (room?.t === 'l') {
```
Comment (murtaza98) [inline app/ui/client/views/app/room.js:893]: @renatobecker I forgot to revert this change in the previous commit. So I'm doing it here

Comment (murtaza98) [reply app/ui/client/views/app/room.js:893]: Since we don't have a `Chat History` button in contacts directory, this is no longer needed

**Review by renatobecker — APPROVED** [review]

**Pull request merged**